common/util.go

```diff
@@ -90,7 +98,7 @@ func HandleHTTPRequest(message string, headers http.Header, data ConnectorMetada
 		return nil, fmt.Errorf("every function invocation retry failed; final retry gave empty response. http_endpoint: %v, source: %v", data.HTTPEndpoint, data.SourceName)
 	}
 
-	if resp.StatusCode != 200 {
+	if resp.StatusCode < 200 && resp.StatusCode > 300 {
```

Comment:
Please raise a separate PR for common package, it would make testing less error-prone.